src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedString = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === this.length - 1 && this[i] === undefined) {
+        break;
+      }
+
+      if (i === this.length - 1) {
+        joinedString += this[i];
+        break;
+      }
```

Comment:
you can redo it and put down to only add separator when it's NOT the last element